### ISSUE TITLE
Decide on NLP 

### ISSUE THREAD

**Issue by walter-scott** [zampolli75/isds1102slackbot#2]
For our purposes, from what I'm gathering either [wit.ai](https://wit.ai) or  [Google's DialogFlow (previously api.ai)](https://dialogflow.com) will suit our needs best. 

Wit.ai seems is a secure choice, as it was apparently [acquired by facebook](https://techcrunch.com/2015/01/05/facebook-wit-ai/) and completely free. According to [this Quora post](https://www.quora.com/What-is-the-basic-difference-between-API-ai-wit-ai-IBM-Watson-and-Luis-ai) (which offers a great comparison of our 5 options) it is best for "simple B2C (business to consumer) bots." 

Google's DialogFlow has a [free plan](https://dialogflow.com/pricing/) that we should not exceed the limits on. 

BY FAR, the best comparison I found is [this article in Medium](https://medium.com/@abraham.kang/understanding-the-differences-between-alexa-api-ai-wit-ai-and-luis-cortana-2404ece0977c). I mainly use it because it sums up things I found from other sources. It talks about how both wit.ai and DialogFlow can "can be used in cases where the platforms learn new commands semantically similar to those input by the developer as triggers for intents." However, the two biggest differences between the two are visualization and invoking of business logic. Apparently wit.ai "provides a developer GUI which includes a visual representation of the conversation flows, business logic invocations, context variables, jumps, and branching logic," something DialogFlow does not. And when it comes to invoking business logic (which I don't completely understand), wit.ai requires you to invoke it yourself where as DialogFlow does not. Although, wit.ai's developer GUI (mentioned above) apparently helps you do this. 

To add another note from the article: DialogFlow has a "Lack of visualization of conversation flows," a problem that wit.ai solves with it's "Stories Feature."Wit.ai also supports "the idea of roles for entities," which DialogFlow does not. Meaning, "When you say “from LA to New York” both LA and New York are location entities but you can further distinguish between a fromLocation (LA) and toLocation (New York)." LUIS is the only other NLP that does this. 

In conclusion, I think we should use wit.ai. At the beggining was biased towards wit.ai because I have already set it up, so I really tried to be critical on it. I was worried about its reliability and its scalability, but after all it appears to be a smart option. It will not only be easy to set up and understand what is going on (through its nice visualization), but we will not be limited in what we can accomplish.

Please read the links and let me know what you think.

**Comment by zampolli75** [comment]
This is great. I went through the links and I agree - Wit is the way to go.
Let's work on it tomorrow.

Nice work :-)
